### PR TITLE
Bluetooth: ISO: Split connected and broadcast ISO Kconfig

### PR DESCRIPTION
**Pull request by Thalley**
Add a new Kconfig option, BT_ISO_UNICAST, to make it possible
either configure unicast only, broadcast only or both.

This results in some code being moved, but not modified, and
should not effect anything.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

**Comment by Thalley** [comment]
Created as a draft PR since 1) we are in a code freeze and I want to avoid making too much noise there and 2) we have some outstanding PRs (especially https://github.com/zephyrproject-rtos/zephyr/pull/33889) that will be affected by this change. 

There's a lot of line changes, but in iso.c it is 99% just moving code. The changes in iso.c is: 1) Creating two prototypes for some unicast functions, and 2) Replacing a check for `if (!iso->acl)` with `if (iso->acl && IS_ENABLED(CONFIG_BT_ISO_UNICAST))`.

File: subsys/bluetooth/Kconfig [hunk 1]
```diff
@@ -224,17 +224,34 @@ endif # BT_CONN
...
+	  Isochronous channels.
 
 if BT_ISO
 
 config BT_ISO_MAX_CHAN
 	int "Maximum number of simultaneous ISO channels"
 	depends on BT_ISO
-	default BT_MAX_CONN
+	default BT_MAX_CONN if BT_CONN
+	default 1
```
Comment (asbjornsabo) [inline subsys/bluetooth/Kconfig:254]: Is it allowed to have two default values?  Which takes presedence?  (It seems the intention is that the first should take precedence.)

File: subsys/bluetooth/host/hci_core.c [hunk 2]
```diff
@@ -2141,11 +2141,12 @@ static const struct event_handler meta_events[] = {
 #endif /* CONFIG_BT_CONN */
 #endif /* defined(CONFIG_BT_PER_ADV_SYNC) */
 #endif /* defined(CONFIG_BT_EXT_ADV) */
-#if defined(CONFIG_BT_ISO)
+#if defined(CONFIG_BT_ISO_UNICAST)
 	EVENT_HANDLER(BT_HCI_EVT_LE_CIS_ESTABLISHED, hci_le_cis_estabilished,
 		      sizeof(struct bt_hci_evt_le_cis_established)),
 	EVENT_HANDLER(BT_HCI_EVT_LE_CIS_REQ, hci_le_cis_req,
 		      sizeof(struct bt_hci_evt_le_cis_req)),
+#endif /* (CONFIG_BT_ISO) */
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/hci_core.c:2149]: CONFIG_BT_ISO_UNICAST

File: subsys/bluetooth/Kconfig (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/Kconfig:254]: Yes it is. I think it's going from top to bottom, or at the very least any `default x if y` takes precedence.

**Comment by Thalley** [comment]
@Vudentz @joerchan @jhedberg I'd like to get this reviewed as it is a blocker for some other changes I have incoming. It's also hard to keep up to date whenever anything changes in iso.c 

The PR doesn't change any functionality, and only regroups pieces of code (of course we don't have any automated tests to fully verify this though).

**Review by jhedberg — APPROVED** [review]
No objections, but I’d have preferred it if you could find a way to split this up into slightly smaller patches

**Comment by Thalley** [comment]
> No objections, but I’d have preferred it if you could find a way to split this up into slightly smaller patches

I agree after seeing the git change list :) Would have been easier to determine if each function (or smaller groups of functions) were moved in smaller patches.

**Review by asbjornsabo — APPROVED** [review]
My comments have been answered/resolved

**Pull request merged**